agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -457,3 +470,70 @@ func (state *DockerTaskEngineState) GetTaskByIPAddress(addr string) (string, boo
 	taskARN, ok := state.ipToTask[addr]
 	return taskARN, ok
 }
+
+// GenerateAndAddV3EndpointID generates a container identifier to be used in v3 metadata endpoint,
+// and update relevant maps. Return false if we fail to generate an unique identifier, otherwise
+// return true
+func (state *DockerTaskEngineState) GenerateAndAddV3EndpointID(task *apitask.Task, container *apicontainer.Container, uuid uuidwrapper.UUID) bool {
```

Comment:
I'd prefer it if the uuid object were either bound to the DockerTaskEngineState struct or stored in a package-level variable.

```golang
// either do something like this...
var uuidFunc UUID

init() {
    uuidFunc = uuidwrapper.NewUUID()
}

// or set it up so you can do something like this (more consistent with other modules in agent):
state.uuid.New()
```

Reply:
if it's a global var, then if i want to test it, i need to add the logic to override it when another uuid provider is specified, which seems tricky, because the logic is specially for testing.. i think passing a uuid provider around is fine since its structure is very simple (empty struct for dynamic provider, one string field for static provider)